packages/e2e-tests/specs/experiments/multi-entity-saving.test.js

```diff
@@ -153,6 +157,28 @@ describe( 'Multi-entity save flow', () => {
 	} );
 
 	describe( 'Site Editor', () => {
+		// Site Editor specific cleanup.
+		afterAll( async () => {
+			// Delete uninitiated template part.
+			const [ uninitiatedTemplatePart ] = await page.$x(
+				uninitiatedTemplatePartSelector
+			);
+			await uninitiatedTemplatePart.click();
+			await page.waitForSelector( selectedTemplatePartSelector );
+			await page.keyboard.press( 'Backspace' );
+
+			// Save again now that it is gone.
+			const enabledButton = await page.waitForSelector(
+				activeSaveSiteSelector
+			);
+			await enabledButton.click();
+
+			const saveButton = await page.waitForSelector(
+				entitiesSaveSelector
+			);
+			await saveButton.click();
+		} );
```

Comment:
so if I understand properly, previously executed tests leave templates and template parts in the database that are used when we access the test again right? can't we just remove them by going to the templates listing under "Appearance"?

Reply:
> so if I understand properly, previously executed tests leave templates and template parts in the database that are used when we access the test again right? can't we just remove them by going to the templates listing under "Appearance"?

We are doing that as part of the `afterAll` cleanup here:  going to the template and template part listings under 'appearance' and removing them all.

Inside the Site Editor, if we have added a template part and then delete that template part from the listings in "appearance", it will not remove that template part from the site entity.  The site editor will load up with that template part in place still, and if you re-save everything it will re-save into 'appearance' as well.

That is why we are currently avoiding fully inserting fully template parts with edits to save in the site editor test.  Also, we don't need to do those things.  So now we only add the Template Part placeholder which is enough to dirty the site entity to allow us to trigger the save functionality, then we remove the placeholder as cleanup.

Reply:
>  If we were to move this to core-data, these would go along with it and Im assuming that would be fine? Or is there some other issue here?

If we remove both the templates and the template parts, won't we come back to the starting state (just theme files)?

Reply:
I thought that would be the case as well, but not completely?  Try going into the site editor, adding a template part, adding some content to it, and saving.  Then goto appearance and delete all your template parts.  Return to the site editor and notice the site editor still knows the template parts slugs and content, and is ready to save them all since it cannot find them in the database anymore.

Reply:
> Then goto appearance and delete all your template parts

I'm saying delete not just the template parts but also the templates, did you try that?

Reply:
Oh, yes.. thank you!  I thought I was removing templates as well, but it turns out `trashExistingPosts` wasn't functioning properly for template and template parts.   It was adding the query arg as `postType` as opposed to `post_type`.  So now we have an underscore in a key name in JavaScript to fix that 🤣 

```	
	const query = addQueryArgs( '', {
		post_type: postType,
	} ).slice( 1 );
```
A bit against standard JS convention, but it gets us to the right page now.  Do you see any potential issues with having the underscore here?

Now this cleanup is working as expected, so I removed that unnecessary cleanup function.

Reply:
underscore is ok there.